src/pykeen/nn/modules.py

```diff
@@ -1090,3 +1091,21 @@ def _prepare_hrt_for_functional(
         t: TailRepresentation,
     ) -> MutableMapping[str, torch.FloatTensor]:  # noqa: D102
         return dict(h=h[0], h_inv=h[1], r=r[0], r_inv=r[1], t=t[0], t_inv=t[1])
+
+
+class PairREInteraction(TranslationalInteraction[FloatTensor, Tuple[FloatTensor, FloatTensor], FloatTensor]):
+    """A stateful module for the PairRE interaction function.
+
+    .. seealso:: :func:`pykeen.nn.functional.pair_re_interaction`
+    """
+
+    relation_shape = ("d", "d")
+    func = pkf.pair_re_interaction
+
+    @staticmethod
+    def _prepare_hrt_for_functional(
+        h: HeadRepresentation,
+        r: RelationRepresentation,
```

Comment:
I think you can put the typing here directly as `Tuple[FloatTensor, FloatTensor]` since the definition of the class already assigns these values to the generics